inductiva/resources/machines_base.py

```diff
@@ -132,20 +133,72 @@ def from_api_response(cls, resp: dict):
         machine_group = cls(
             machine_type=resp["machine_type"],
             data_disk_gb=resp["disk_size_gb"],
-            provider=resp["provider_id"],
             register=False,
         )
         machine_group._id = resp["id"]
+        machine_group.provider = resp["provider_id"]
         machine_group._name = resp["name"]
         machine_group.create_time = resp["creation_timestamp"]
         machine_group._started = True
 
         return machine_group
 
-    def start(self, **kwargs):
+    def update_termination_timers(self,
+                                  max_idle_time: float = None,
+                                  auto_terminate: Union[str, float] = None):
+        """Helper function to update the termination timers of a machine group.
+
+        At the moment, this serves as an helper function, but it could also be
+        provided to the users on demand.
+
+        Args:
+            max_idle_time (float): Time in minutes that the machine can remain
+                idle.
+            auto_terminate (float, str): Time to automatically terminate the
+                machines independently of any simulations running there.
+                The time can be a float, indicating the number of hours the
+                machine up until the machine can be up, or an actual timestamp
+                with the format '2024-12-31T00:00:00+00'.
+        """
+
+        # Convert max_idle_time from minutes to seconds
+        if max_idle_time is not None:
+            max_idle_time = max_idle_time * 60
+
+        # Convert float auto_terminate_ts to ISO format
+        if isinstance(auto_terminate, float):
+            auto_terminate = datetime.datetime.now(datetime.timezone.utc) + \
+                datetime.timedelta(hours=auto_terminate)
+            auto_terminate = auto_terminate.isoformat()
```

Comment:
having `auto_terminate` be a `datetime` object will largely simplify this logic.
You'll still have to check that `auto_terminate > now`!